src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -343,5 +366,37 @@ public int MultiDimensionalArrayRank
                 return (int)((BaseSize - (uint)(3 * sizeof(IntPtr))) / (uint)(2 * sizeof(int)));
             }
         }
+
+        public MethodTable* BaseMethodTable
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                int mtFieldOffset = 0x10 / sizeof(nuint);
```

Comment:
Why this cannot be just a regular field access? The ifdefs can be on the field offset.

Reply:
There would be 3 combinations for BaseMethodTable and 4 combinations for the InterfaceMap (32/64bit X Ret/Debug)

---
In reply to: [368260513](https://github.com/dotnet/runtime/pull/1068#discussion_r368260513) [](ancestors = 368260513)

Reply:
Use `const` + ifdefs to make it look pretty?

Reply:
I will try, but I think it might look worse than this.

---
In reply to: [368261592](https://github.com/dotnet/runtime/pull/1068#discussion_r368261592) [](ancestors = 368261592,368260513)

Reply:
Use `const` + ifdefs to make it look pretty?

Reply:
that is the idea. These can be constant expressions, I think.

---
In reply to: [368262114](https://github.com/dotnet/runtime/pull/1068#discussion_r368262114) [](ancestors = 368262114)